src/btree/bt_curnext.c

```diff
@@ -485,10 +486,11 @@ __cursor_key_order_check_col(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, boo
     }
 
     WT_RET(__wt_msg(session, "dumping the cursor page"));
-    WT_RET(__wt_debug_cursor_page(&cbt->iface, NULL));
+    WT_WITH_PAGE_INDEX(session, ret = __wt_debug_dump_tree(&cbt->iface));
     WT_RET_PANIC(session, EINVAL,
       "WT_CURSOR.%s out-of-order returns: returned key %" PRIu64 " then key %" PRIu64,
       next ? "next" : "prev", cbt->lastrecno, cbt->recno);
+    return (ret);
```

Comment:
Will this `return` be executed if we have `WT_RET_PANIC` just before?

Reply:
Fixed in 548cc0338987afb463397e5b10e6ebaf1e789f75

Reply:
`WT_RET_PANIC` sounds good.

Reply:
Hmm, I still think we can get rid of the `err` section and use `WT_RET_PANIC` instead of `WT_ERR_PANIC`. We can discuss this offline.

Reply:
Since `EINVAL` is given to `WT_ERR_PANIC`, it will jump to the `err` label, that's why we need it.

Reply:
Hmm, This is a diagnostic code. `__wt_panic` dumps core in diagnostic builds. We are not returning an error to the user here once we detected an out of order. Am I missing anything?

Reply:
Good point, since it is diagnostic code the program aborts, you are right. We are then coming back to my first comment, the `return` statement becomes useless, or am I mistaken?
I'd simply return `__wt_panic` if that's possible line 492.

Reply:
The reason we have an error label is that we save the return value from `__wt_debug_tree_all ` function and hopefully if it returns success, then it does not makes sense to call WT_RET_PANIC with ret (success) and if we pass EINVAL in `WT_RET_PANIC` then s_all will complain that the ret is being assigned a value and not returned.

Reply:
ok, understood. It is to suppress the s_all. Can you add a comment to explain the same, otherwise use WT_UNUSED for the `ret`.

Reply:
Fixed in a252fbaf489c758069fda73df39ff7cbf33e4512